### PR TITLE
fix(Scripts/BlackMorass): Implemented wipe event.

### LINKED ISSUE #14856: [Opening the Dark Portal] Wipe Event is incorrect
### Current Behaviour

Medivh drops dead instantly without animation once 0% integrity is reached.
Black Portal Dummy never despawns.

### Expected Blizzlike Behaviour

On 0% reached:

Play Sound 10441
Text: No! Damn this feeble, mortal coil! (14)
Wait 4000ms
Medivh Dies, performing the death animation
All Medivh Auras removed
All Entries 18625, 21862, 17838 Despawn
Wait 2000ms
All Adds cast spell 7791 on self and despawn 1200ms later

### Source

_No response_

### Steps to reproduce the problem

`.go c id 15608`
Let the adds destroy Medivh's Shield

### Extra Notes

_No response_

### AC rev. hash/commit

https://github.com/azerothcore/azerothcore-wotlk/commit/c3dd1b7a5ca0a9945d609213fdd3c4a19e0106e2

### Operating system

Windows 10

### Custom changes or Modules

_No response_

### PR DESCRIPTION
**Pull request by UltraNix**
Fixes #14856

## Issues Addressed:

- Closes #14856

## Tests Performed:

- Tested ingame.

## How to Test the Changes:

`.go c id 15608`
Let the adds destroy Medivh's Shield

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Maelthyrr** [comment]
The PR work but the effect are wrong:

Without: 

WoWScrnShot_020423_202740

With:

WoWScrnShot_020423_193514

File: src/server/scripts/Kalimdor/CavernsOfTime/TheBlackMorass/instance_the_black_morass.cpp [hunk 1]
```diff
@@ -308,6 +342,54 @@ class instance_the_black_morass : public InstanceMapScript
                 case EVENT_SUMMON_KEEPER:
                     SummonPortalKeeper();
                     break;
+                case EVENT_WIPE_1:
+                    if (Creature* medivh = instance->GetCreature(_medivhGUID))
+                    {
+                        medivh->RemoveAllAuras();
+                    }
+                    Events.ScheduleEvent(EVENT_WIPE_2, 500);
```
Comment (Kitzunu) [inline src/server/scripts/Kalimdor/CavernsOfTime/TheBlackMorass/instance_the_black_morass.cpp:350]: chrono

File: src/server/scripts/Kalimdor/CavernsOfTime/TheBlackMorass/instance_the_black_morass.cpp [hunk 2]
```diff
@@ -187,20 +199,42 @@ class instance_the_black_morass : public InstanceMapScript
...
                             {
-                                Unit::Kill(medivh, medivh);
+                                medivh->SetImmuneToNPC(true);
+
+                                if (medivh->IsAIEnabled)
+                                {
+                                    medivh->AI()->Talk(SAY_MEDIV_DEATH);
+                                }
+
+                                Events.ScheduleEvent(EVENT_WIPE_1, 4000);
```
Comment (Kitzunu) [inline src/server/scripts/Kalimdor/CavernsOfTime/TheBlackMorass/instance_the_black_morass.cpp:223]: same here

**Comment by UltraNix** [comment]
@Maelthyrr Should be okay now.

File: src/server/scripts/Kalimdor/CavernsOfTime/TheBlackMorass/the_black_morass.h [hunk 3]
```diff
@@ -75,19 +75,50 @@ enum CreatureIds
...
 };
 
+enum medivhSays
+{
+    SAY_MEDIV_ENTER                     = 0,
+    SAY_MEDIV_DEATH                     = 5,
+    SAY_MEDIV_WIN                       = 6,
+    SAY_MEDIV_ORCS_ENTER                = 7,
+
+    SAY_MEDIV_ORCS_ANSWER               = 0
```
Comment (Gultask) [inline src/server/scripts/Kalimdor/CavernsOfTime/TheBlackMorass/the_black_morass.h:106]: Wouldn't it be better to write MEDIVH instead of MEDIV?

**Review by Gultask — COMMENTED** [review]
Just a little comment. Another interesting thing:
NPC_DP_EMITTER_STALKER doesn't exist in any sniffs I have. Neither Classic or Retail full runs, wipe or full clears.

**Review by IntelligentQuantum — APPROVED** [review]

**Comment by Kitzunu** [comment]
Cpnflicts

**Comment by UltraNix** [comment]
@Kitzunu Done.

**Pull request merged**